playground/server.py

```diff
@@ -30,28 +30,35 @@ def play():
 
     if regex_digit.match(x) is None:
         return {"code": 400, "message": "illegal input in x"}, 400
-    elif regex_digit.match(y) is None:
+
+    if regex_digit.match(y) is None:
         return {"code": 400, "message": "illegal input in y"}, 400
-    elif regex_color.match(color) is None:
+
+    if regex_color.match(color) is None:
         return {"code": 400, "message": "illegal input in stone"}, 400
-    else:
-        if referee.turn_check(board.log) == color:
-            stone = Stone(x, y, color)
-            if referee.valid_check(stone, board.log):
-                board.print_on_playing()
-                board.put_stone(stone)
-                if referee.end_check(board.log) != "keep play":
-                    board.print_winner(color)
-                board.save_figure()
-                return render_template("board.html")
-            else:
-                board.print_illegal_stone()
-                board.save_figure()
-                return render_template("board.html")
-        else:
-            board.print_illegal_turn()
-            board.save_figure()
-            return render_template("board.html")
+
+    if color != referee.turn_check(board.log):
+        board.print_illegal_turn()
+        board.save_figure()
+        return render_template("board.html")
+
+    stone = Stone(x, y, color)
+    if not referee.valid_check(stone, board.log):
+        board.print_illegal_stone()
+        board.save_figure()
+        return render_template("board.html")
+
+    board.put_stone(stone)
+    end_check = referee.end_check(board.log, board.config)
+    if end_check.is_end:
+        winner = "no one" if end_check.is_tie else color
+        board.print_winner(winner)
+        board.save_figure()
+        return render_template("board.html")
+
+    board.print_on_playing()
+    board.save_figure()
+    return render_template("board.html")
 
```

Comment:
code block
```
board.save_figure()
return render_template("board.html")
```
is repeated every time from :40 to :61. Is there any way to simplify this?

Reply:
I tried exactly same thing but I end up with this as a minimum repeatation